Modules/_threadmodule.c

```diff
@@ -1137,13 +1137,8 @@ information about locks.");
 static PyObject *
 thread_get_ident(PyObject *self)
 {
-    long ident;
-    ident = PyThread_get_thread_ident();
-    if (ident == -1) {
-        PyErr_SetString(ThreadError, "no current thread ident");
-        return NULL;
-    }
-    return PyLong_FromLong(ident);
+    unsigned long ident = PyThread_get_thread_ident();
```

Comment:
In _PyImport_ReleaseLock(), you test "== PYTHREAD_INVALID_THREAD_ID". Please be consistent: remove the test in _PyImport_ReleaseLock(), or restore the test it. I'm in favor of dropping the test: pthread_self() and GetCurrentThreadId() cannot fail.